libs/admin-panel/src/lib/components/dependency-modal.tsx

```diff
@@ -0,0 +1,145 @@
+'use client';
+
+import { FormEvent, useEffect, useState } from 'react';
+
+import { App, Version } from '@prisma/client';
+
+import { Button } from '@verity/ui/button';
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@verity/ui/dialog';
+import { Label } from '@verity/ui/label';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@verity/ui/select';
+
+export interface DependencyModalProps {
+  readonly currentAppId: string;
+  readonly onFormSubmit: (versionId: number) => void;
+}
+
+export function DependencyModal({ currentAppId, onFormSubmit }: DependencyModalProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [appId, setAppId] = useState<string | null>(null);
+  const [version, setVersion] = useState<Version | null>(null);
+  const [appsIds, setAppsIds] = useState<string[]>([]);
+  const [versions, setVersions] = useState<Version[]>([]);
+
+  useEffect(() => {
+    resetState();
+
+    const updateAppsIds = async () => {
+      try {
+        const response = await fetch('api/apps');
+        const data: App[] = await response.json();
+
+        const ids = data.map(({ id }) => id).filter((id) => id !== currentAppId);
+        setAppsIds(ids);
+      } catch (error) {
+        console.error('Error:', error);
+      }
+    };
+
+    if (isOpen) {
+      void updateAppsIds();
+    }
+  }, [isOpen]);
+
+  useEffect(() => {
+    const updateVersions = async () => {
+      try {
+        const response = await fetch(`api/apps/${appId}/versions`);
+        const data: Version[] = await response.json();
+
+        setVersions(data);
+      } catch (error) {
+        console.error('Error:', error);
+      }
+    };
+
+    if (appId) {
+      void updateVersions();
+    }
+  }, [appId]);
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (!version?.id) return;
+    onFormSubmit(version.id);
+    setIsOpen(false);
+  };
+
+  const handleVersionChange = (id: number) => {
+    const version = versions.find((version) => version.id === id) ?? null;
+    setVersion(version);
+  };
+
+  const resetState = () => {
+    if (!isOpen) {
```

Comment:
I would prefer to move the condition inside the ﻿useEffect, as it currently requires scrolling to understand that we don't actually reset the state every time ﻿isOpen changes.